lib/services/hook.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+var http =  require('http');
+var https =  require('https');
+var url = require('url');
+
+module.exports = hookServiceFactory;
+
+hookServiceFactory.$provide = 'Services.Hook';
+hookServiceFactory.$inject = [
+    'Services.Messenger',
+    '_',
+    'Assert',
+    'Services.Waterline'
+];
+
+function hookServiceFactory(
+    messenger,
+    _,
+    assert,
+    waterline
+) {
+    function HookService(){
+    }
+    
+    /**
+     * Publishes the JSON serialized from 'data' into AMQP based on exchange and routingKey.
+     *
+     * @param {string} name
+     * @param {string} routingKey
+     * @param {Object} data
+     * @param {Object} options
+     * @returns {Promise}
+     */
+    HookService.prototype.publish = function(name, routingKey, data, options) {
```

Comment:
The AMQP messenger logic is put within the hook, but actually they are in equal status, so you should remove all AMQP messenger code out of the hookService.

@sunnyqianzhang has a PR to split internal and external events, bases on her PR, I think the better place to combine the hook and AMQP messenger is "Services.Messenger"